### PR TITLE
fix: action Improvements - overflow error, messages, +

### PR DESCRIPTION
**Pull request by StuartRucker**
Moved the processing of upload metadata to the action from the server so the endpoint is more versatile and avoids overflow error. In conjunction with https://github.com/GoogleCloudPlatform/flaky-service/pull/223

- Fixes https://github.com/GoogleCloudPlatform/flaky-service/issues/132 by changing route to `/build/gh/v1` which creates a plan for future updates when there are users. A new endpoint will be created, without removing the old, that the newest version of the action can access
- Fixes https://github.com/GoogleCloudPlatform/flaky-service/issues/219 by not passing in the entire `github` context as environment variable
- https://github.com/GoogleCloudPlatform/flaky-service/issues/93 Addresses but does not necessarily "fix" this issue by adding more informative error messages that use the `core.warning` feature. Fails if action is used blatantly incorrectly. Warns if there is an issue such as an invalid token (so the rest of the steps in the action can continue)
- Uses environment variables when possible instead of passing in parameters

- Deletes an attempt at https://github.com/GoogleCloudPlatform/flaky-service/issues/222 (checking if repo is private will be done server side with Github token in the future)

### CODE REVIEW

File: packages/action/index.js [hunk 1]
```diff
@@ -22,32 +22,78 @@ const core = require('@actions/core');
...
   } catch (error) {
-    console.log(error);
+    core.warning(error);
     core.setFailed(error.message);
   }
 }
 
 
-main();
+main();
```
Comment (tbpg) [inline packages/action/index.js:99]: Newline at the end.

File: packages/action/index.js [hunk 2]
```diff
@@ -22,32 +22,78 @@ const core = require('@actions/core');
 async function main() {
   try {
     // `who-to-greet` input defined in action metadata file
+    if(!core.getInput('repo-token') || !core.getInput('os') || !core.getInput('file-path') || !core.getInput('repo-token') ){
```
Comment (tbpg) [inline packages/action/index.js:25]: Looks like `repo-token` is checked twice?

File: packages/action/index.js [hunk 3]
```diff
@@ -22,32 +22,78 @@ const core = require('@actions/core');
...
+      throw "Github action missing required fields. Refer to documentation at https://github.com/googlecloudplatform/flaky-service";
```
Comment (tbpg) [inline packages/action/index.js:26]: Can be a separate PR, but it doesn't look like the root README has relevant docs.

File: packages/action/index.js [hunk 4]
```diff
@@ -22,32 +22,78 @@ const core = require('@actions/core');
...
+    }
+
+   
+    let buildmessageStr = process.env.GITHUB_WORKFLOW + ' ' + process.env.GITHUB_RUN_NUMBER;
```
Comment (bcoe) [inline packages/action/index.js:30]: Nice, I'm glad we're building up this human readable string now.

File: packages/action/index.js [hunk 5]
```diff
@@ -22,32 +22,78 @@ const core = require('@actions/core');
...
+      description: core.getInput('repo-description'),
+      buildmessage: buildmessageStr,
+      token: core.getInput('repo-token')
     };
-    const fileType = core.getInput('logtype');
-    console.log('reading from: ' + core.getInput('filepath'));
-    const data = fs.readFileSync(
-        core.getInput('filepath'), 'utf8');
 
+    metaData.matrix = JSON.stringify(metaData.environment.matrix, Object.keys(metaData.environment.matrix).sort()); //consistancy
```
Comment (bcoe) [inline packages/action/index.js:55]: curious what the reasoning is for sending this to the API as a serialized string, vs., just leaving it an array. What's the ultimate representation in the database?

File: packages/action/index.js [hunk 6]
```diff
@@ -22,32 +22,78 @@ const core = require('@actions/core');
...
       method: 'POST',
       body: JSON.stringify(sendMe),
       headers: {'Content-Type': 'application/json'},
     });
     const outcomeText = await outcome.text();
-    console.log('\nSERVER RESPONSE:');
-    console.log(outcomeText);
+    const outcomeAsJSON = JSON.parse(outcomeText);
+    if(outcomeAsJSON.error){
+      core.warning("Upload Failed - Status " + outcome.status);
```
Comment (bcoe) [inline packages/action/index.js:80]: if we get an error, we should probably:

```
process.exitCode = 1
```

☝️ this will mean that the action indicates a failure.

File: packages/action/index.js (hunk 5 above)
Comment (StuartRucker) [reply packages/action/index.js:55]: In the database, its stored as a string. It keys need to be in alphabetic order because  in order to have a dropdown which filters by filter, you need to check for equality between the matrices. This could be done differently though. Another motivation for storing it as a string is to avoid a user input to be a Firestore key.

File: packages/action/index.js (hunk 6 above)
Comment (StuartRucker) [reply packages/action/index.js:80]: Okay sounds good. I initially avoided this because it could be annoying if your entire action gets cancelled because we are having server issues.
